src/ComponentModel.cpp

```diff
@@ -733,6 +733,14 @@ std::string xtypes::ComponentModel::export_to_basic_model()
         }
     
     }
+    this->set_unknown_fact_empty("configured_for");
```

Comment:
This line does not make much sense. You should use has_facts() to check whether the facts are present.